### PR TITLE
Add FAQ entry for running Cassandra

### PR DESCRIPTION
**Pull request by mbanikazemi**
Addresses issue istio/istio#10053

Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: content/en/faq/applications/cassandra.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+Cassandra configuration directory (e.g. `/etc/cassandra`).  There are
+various startup scripts (and yaml files) used for starting Cassandra and care
+should be given to how these parameters are set by these scripts. As an example, an
+script used to configure and start Cassandra uses the value of the environment variable
+`CASSANDRA_LISTEN_ADDRESS` for setting `listen_address`.
+
+1. Service Definition
+
+Another issue to keep in mind is having ports specified in the
+Cassandra's headless service as shown below:
```
[the remark this answers is not in the document]
Comment (mbanikazemi) [reply content/en/faq/applications/cassandra.md]: Done.

File: content/en/faq/applications/cassandra.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
+---
+title: Can I run Casandra inside an Istio mesh?
+description: How to run Cassandra with Istio.
+weight: 50
+keywords: [cassandra]
+---
+
+The sort answer is yes, Cassandra can run inside an Istio mesh.
```
[the remark this answers is not in the document]
Comment (mbanikazemi) [reply content/en/faq/applications/cassandra.md]: Done.

File: content/en/faq/applications/cassandra.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    name: tls-intra-node
+  - port: 7199
+    name: jmx
+  - port: 9042
+    name: cql
+  selector:
+    app: cassandra
+{{< /text >}}
+
+1. Using MTLS
```
[the remark this answers is not in the document]
Comment (mbanikazemi) [reply content/en/faq/applications/cassandra.md]: Done.

**Comment by geeknoid** [comment]
You have a few linting errors:

./content/en/faq/applications/cassandra.md:8: MD009 Trailing spaces
./content/en/faq/applications/cassandra.md:17: MD012 Multiple consecutive blank lines

File: content/en/faq/applications/cassandra.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+The short answer is yes, Cassandra can run inside an Istio mesh.
+
+By default, Cassandra broadcasts the address it uses for binding
+(accepting connections) to other Cassandra nodes as its address. This
+is usually the pod IP address and works fine without a service
+mesh. However, with a service mesh this configuration does not
+work. Istio and other service meshes require the `localhost`
+(`127.0.0.1`) to be the address for binding.
+
+1. Cassandra Configuration
```
[the remark this answers is not in the document]
Comment (mbanikazemi) [reply content/en/faq/applications/cassandra.md]: Done.

**Comment by geeknoid** [comment]
@rshriram Could you take a look at this?

File: content/en/faq/applications/cassandra.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+description: How to run Cassandra with Istio.
+weight: 50
+keywords: [cassandra]
+---
+
+By default, Cassandra broadcasts the address it uses for binding
+(accepting connections) to other Cassandra nodes as its address. This
+is usually the pod IP address and works fine without a service
+mesh. However, with a service mesh this configuration does not
+work. Istio and other service meshes require the `localhost`
```
Comment (frankbu) [inline content/en/faq/applications/cassandra.md]:
```suggestion
work. Istio and other service meshes require `localhost`
```

File: content/en/faq/applications/cassandra.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+(`127.0.0.1`) to be the address for binding.
+
+## Cassandra Configuration
```
Comment (frankbu) [inline content/en/faq/applications/cassandra.md]:
```suggestion
```

File: content/en/faq/applications/cassandra.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
```
Comment (frankbu) [inline content/en/faq/applications/cassandra.md]:
```suggestion
```

File: content/en/faq/applications/cassandra.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+There are two configuration parameters to pay attention to:
+[`listen_address`](http://cassandra.apache.org/doc/latest/configuration/cassandra_config_file.html?highlight=listen_address#listen-address)
+and
+[`broadcast_address`](http://cassandra.apache.org/doc/latest/configuration/cassandra_config_file.html?highlight=listen_address#broadcast-address). For
+running Cassandra in an Istio mesh they need to be set appropriately.
```
Comment (frankbu) [inline content/en/faq/applications/cassandra.md]:
```suggestion
running Cassandra in an Istio mesh,
```

File: content/en/faq/applications/cassandra.md [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+The `listen_address` parameter should be set to `127.0.0.1` and the
```
Comment (frankbu) [inline content/en/faq/applications/cassandra.md]:
```suggestion
the `listen_address` parameter should be set to `127.0.0.1` and the
```

File: content/en/faq/applications/cassandra.md [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+`broadcast_address` parameter should be set to the pod IP address.
+
+These configuration parameters are defined in `cassandra.yaml` in the
+Cassandra configuration directory (e.g. `/etc/cassandra`).  There are
+various startup scripts (and yaml files) used for starting Cassandra
+and care should be given to how these parameters are set by these
+scripts. As an example, an script used to configure and start
+Cassandra uses the value of the environment variable
```
Comment (frankbu) [inline content/en/faq/applications/cassandra.md]:
```suggestion
Cassandra use the value of the environment variable
```

File: content/en/faq/applications/cassandra.md [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+`CASSANDRA_LISTEN_ADDRESS` for setting `listen_address`.
+
```
Comment (frankbu) [inline content/en/faq/applications/cassandra.md]: Isn't the service port requirement gone in 1.3? If so, remove the following.

File: content/en/faq/applications/cassandra.md [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+[`broadcast_address`](http://cassandra.apache.org/doc/latest/configuration/cassandra_config_file.html?highlight=listen_address#broadcast-address). For
+running Cassandra in an Istio mesh they need to be set appropriately.
+The `listen_address` parameter should be set to `127.0.0.1` and the
+`broadcast_address` parameter should be set to the pod IP address.
+
+These configuration parameters are defined in `cassandra.yaml` in the
+Cassandra configuration directory (e.g. `/etc/cassandra`).  There are
+various startup scripts (and yaml files) used for starting Cassandra
+and care should be given to how these parameters are set by these
+scripts. As an example, an script used to configure and start
```
Comment (frankbu) [inline content/en/faq/applications/cassandra.md]:
```suggestion
scripts. For example, some scripts used to configure and start
```

File: content/en/faq/applications/cassandra.md (hunk 5 above)
Comment (mbanikazemi) [reply content/en/faq/applications/cassandra.md]: Done.

File: content/en/faq/applications/cassandra.md (hunk 8 above)
Comment (mbanikazemi) [reply content/en/faq/applications/cassandra.md]: Done.

File: content/en/faq/applications/cassandra.md (hunk 9 above)
Comment (mbanikazemi) [reply content/en/faq/applications/cassandra.md]: Done.

File: content/en/faq/applications/cassandra.md (hunk 10 above)
Comment (mbanikazemi) [reply content/en/faq/applications/cassandra.md]: Done.

File: content/en/faq/applications/cassandra.md (hunk 12 above)
Comment (mbanikazemi) [reply content/en/faq/applications/cassandra.md]: Done.

File: content/en/faq/applications/cassandra.md (hunk 7 above)
Comment (mbanikazemi) [reply content/en/faq/applications/cassandra.md]: Done.

File: content/en/faq/applications/cassandra.md (hunk 11 above)
Comment (mbanikazemi) [reply content/en/faq/applications/cassandra.md]: Done.

**Pull request merged**